app/models/host/managed.rb

```diff
@@ -336,6 +350,10 @@ def provisioning_template(opts = {})
     opts[:hostgroup_id]       ||= hostgroup_id
     opts[:environment_id]     ||= environment_id
 
+    host_facets.each do |facet|
+      opts = facet.provisioning_template_options(opts)
```

Comment:
If `provisioning_template_options` returns a Hash, then you would call `opts.merge!(facet.provisioning_template_options)` here